### ISSUE TITLE
Register types treats "const char *" and "char const *" differently

### ISSUE THREAD

**Issue by mattdurak** [Azure/umock-c#129]
Would be nice if either of these mockable functions worked "out of the box"

When setup like this:
```c
#include "umock_c/umocktypes_charptr.h"

TEST_SUITE_INITIALIZE(suite_init)
{
    // ...
    result = umocktypes_charptr_register_types();
    ASSERT_ARE_EQUAL(int, 0, result, "umocktypes_charptr_register_types");
    // ...
}

TEST_FUNCTION(test1)
{
    STRICT_EXPECTED_CALL(mocked_popen("foo", "bar"));
    mocked_popen("foo", "bar");
    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
}
```

This works:
```c
MOCKABLE_FUNCTION(, FILE*, mocked_popen,
    const char*, command,
    const char*, mode);
```

This falls back to `void*`:
```c
MOCKABLE_FUNCTION(, FILE*, mocked_popen,
    char const*, command,
    char const*, mode);
```

> Expected: [mocked_popen(00007FF6BFF3D9EC,00007FF6BFF3D9AC)], Actual: [mocked_popen(00007FF6BFF3DA3C,00007FF6BFF3D9FC)]

**Maybe a nice improvement to `umocktypename_normalize`**

**Comment by dcristoloveanu** [comment]
Hello @mattdurak!

What a wonderful suggestion. As this project happily accepts contributions feel free to PR it :-).

Cheers,
/Dan

**Comment by mattdurak** [comment]
Fixed by #134
